### PR TITLE
Clearly differentiate running elevated vs. drag/drop breaking

### LINKED ISSUE #13928: Run as Admin shortcut + elevate:true -> crash loop
### Windows Terminal version

1.14.2281.0 or preview 1.15.2282.0

### Windows build number

10.0.19044.0

### Other Software

_No response_

### Steps to reproduce

1. Install terminal regardless preview version or  not.
2. Turn on the "Run this profile as Administrator" in Profile->default
3. Restart the terminal.

### Expected Behavior

_No response_

### Actual Behavior

https://user-images.githubusercontent.com/53345636/188541035-ff4005a1-dbfb-4ed7-8c34-3e9f8367c068.mp4

I also tried run terminal as administrator,but it seemed didn't work.

### PR DESCRIPTION
**Pull request by zadjii-msft**
Credit where credit is due - @jboelter did literally all the hard work.

I just separated this out to two elements:
* Are we running elevated?
* Can we drag drop?

As we learned in #7754, the builtin administrator _can_ drag drop. But critically, they are also running as admin! The way we had this logic before, we're treat them as unelevated, because we had been overloading the meaning here.

This splits these into two separate functions. Comes with the added benefit of re-adding the elevation shield to the Terminal window for users with UAC disabled (which was missing before) (and can _still_ be disabled).

Closes #13928

Tested on a Win10 VM with `EnableLua=0`

### CODE REVIEW

File: src/types/utils.cpp [hunk 1]
```diff
@@ -670,8 +679,28 @@ bool Utils::IsElevated()
...
+                // They can not NOT drag drop -> they _can_ drag drop
             }
 
+            // If they are running admin, they cannot drag drop.
+            return wil::test_token_membership(nullptr, SECURITY_NT_AUTHORITY, SECURITY_BUILTIN_DOMAIN_RID, DOMAIN_ALIAS_RID_ADMINS);
+        }
+        catch (...)
+        {
+            LOG_CAUGHT_EXCEPTION();
+            return false;
```
Comment (DHowett) [inline src/types/utils.cpp:691]: should we err on the side of "isDragDropBroken = true" if part of this process explodes?

**Review by DHowett — CHANGES_REQUESTED** [review]

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 2]
```diff
@@ -163,13 +163,26 @@ namespace winrt::TerminalApp::implementation
...
-        return isElevated;
+        return isRunningElevated;
+    }
+    bool TerminalPage::CanDragDrop() const noexcept
+    {
+        static const auto canDragDrop = []() {
+            try
+            {
+                return Application::Current().as<TerminalApp::App>().Logic().CanDragDrop();
+            }
```
[the remark this answers is not in the document]
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/TerminalPage.cpp:180]: Meh, that's what I get for copypasta'ing the IsElevated one

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 3]
```diff
@@ -173,16 +173,12 @@ namespace winrt::TerminalApp::implementation
...
-                return Application::Current().as<TerminalApp::App>().Logic().CanDragDrop();
-            }
-            CATCH_LOG();
-            return true;
-        }();
-
-        return canDragDrop;
+        try
+        {
+            return Application::Current().as<TerminalApp::App>().Logic().CanDragDrop();
```
Comment (DHowett) [inline src/cascadia/TerminalApp/TerminalPage.cpp:178]: same sticking point about isRunningElevated above ;)

**Comment by NyaMisty** [comment]
Thanks God! Finally someone spotted this!
I've just revesed engineered win32k stuffs and experimented a lot with UIPI.
I was just going to say "New Notepad can drag&drop files perfectly, so why WT can't?", then I found this PR.

But my tests on Windows 11 shows that drag&drop file is still not working (with EnableLUA=0)

**Review by DHowett — APPROVED** [review]

**Comment by DHowett** [comment]
(I chose to merge without seeing your comments -- @zadjii-msft, mind tracking them?)

**Pull request merged**